substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/pecoff/cv/CVSymbolSubsectionBuilder.java

```diff
@@ -152,6 +190,86 @@ private void buildFunction(CompiledMethodEntry compiledEntry) {
         addLineNumberRecords(compiledEntry);
     }
 
+    void addLocals(CompiledMethodEntry primaryEntry) {
+        final Range primaryRange = primaryEntry.getPrimary();
+        /* The name as exposed to the linker. */
+        final String externalName = primaryRange.getSymbolName();
+
+        /* Add register parameters - only valid for the first instruction or two. */
+
+        MethodEntry method = primaryRange.getMethodEntry();
+        int gpRegisterIndex = 0;
+        int fpRegisterIndex = 0;
+
+        /* define 'this' as a local just as we define other object pointers */
+        if (!Modifier.isStatic(method.getModifiers())) {
+            final TypeEntry thisType = primaryEntry.getClassEntry();
+            final int typeIndex = cvDebugInfo.getCVTypeSection().getIndexForPointer(thisType);
+            addSymbolRecord(new CVSymbolSubrecord.CVSymbolLocalRecord(cvDebugInfo, "this", typeIndex, 1));
+            addSymbolRecord(new CVSymbolSubrecord.CVSymbolDefRangeRegisterRecord(cvDebugInfo, javaGP64registers[gpRegisterIndex], externalName, 0, 8));
+            addSymbolRecord(new CVSymbolSubrecord.CVSymbolDefRangeFramepointerRelFullScope(cvDebugInfo, 0));
+            gpRegisterIndex++;
+        }
+
+        /* define function parameters (p1, p2...) according to the calling convention */
+        for (int i = 0; i < method.getParamCount(); i++) {
+            final TypeEntry paramType = method.getParamType(i);
+            final int typeIndex = cvDebugInfo.getCVTypeSection().addTypeRecords(paramType).getSequenceNumber();
+            final String paramName = "p" + (i + 1);
+            if (typeIndex == T_REAL64 || typeIndex == T_REAL32) {
+                /* floating point primitive */
```

Comment:
These two cases can also be distinguished by first testing for a primitive type (i.e. folding into the branch below) and then checking whether the primitive type entry flags include NUMERIC but omit INTEGRAL.

Reply:
Done.